GVFS/GVFS.Virtualization/Projection/GitIndexProjection.cs

```diff
@@ -1180,10 +1181,13 @@ private void UpdatePlaceholders()
                         {
                             existingPlaceholders?.Remove(folderPlaceholder.Path);
                             this.placeholderDatabase.Remove(folderPlaceholder.Path);
+                            ++deleteFolderPlaceHolder;

```

Comment:
Currently this is counting both folder placeholders that were deleted by VFS4G, and those that did not exist (but VFS4G was trying to delete).  Do you think these counts should be combined?

cc: @kewillford for his thoughts as well

Reply:
@jeschu1 This depends on what we are trying to get out of this logging.  Are we trying to determine what is causing something to be slow by having a count on the folder placeholders?  Or is it so we know an can report on the number of empty folders that the user had?  There are multiple counts here that could be useful. 
1. The number of folder placeholder that we attempt to delete.  This was the cause of performance issues before we added the check for if it was in the projection.
2. The number of folder placeholders that actually get deleted.  I suspect in most cases now this will be equal to the attempts but would be interesting to know when and how often the are not.
3. The number of deletes that are performed on the database.  Again this would be interesting to see how this number compares with the others.

So I guess it all depends on what we will be using these number for in the logging.

Reply:
To add some additional context here, I originally filed the issue because I found it difficult to figure where time was being spent during projection changes (and what VFS4G was doing).

I think the more details we can include the better, it will help with understanding what VFS4G was doing during a projection change.

Reply:
I'm wondering if a better place for this data would be [here](https://github.com/microsoft/VFSForGit/blob/c6ab22dde56d91591b02026860ccb3e9990d6560/GVFS/GVFS.Common/GVFSLock.cs#L332).  @kewillford i'll get in touch regarding.

Reply:
> I'm wondering if a better place for this data would be here

Is that data currently written to the log file as well?

Reply:
> Is that data currently written to the log file as well?

It looks like it should be.  If it isn't I'll keep this as standalone telemetry.

> RemoveFolderPlaceholderIfEmpty

Do you think we should pull this apart to distinguish when this.fileSystemVirtualizer.DeleteFile gets OK versus FileOrPathNotFound.  @kewillford didnt' have strong feelings here so let me know what you think @wilbaker

Reply:
@jeschu1 I think it would be worth distinguishing them.  Could we update `RemoveFolderPlaceholderIfEmpty` to return an enum value, and then use that to decide which count to increment?

Reply:
@wilbaker @kewillford please give this a fresh look

@wilbaker I confirmed the added values go to the log